tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,31 @@
+"""Fill in the Gaps."""
+import os
+
+
+def fill_gaps(folder_path, prefix):
+    """Fill in the Gaps."""
+    # init file list with first not needed element
+    file_list = [None]
+    os.chdir(folder_path)
+    # get list of tuples of filenames and extentions
+    with os.scandir(folder_path) as scandir:
```

Comment:
Please notice that according to [docs](https://docs.python.org/3/library/os.html#os.scandir) the `os.scandir` does not guarantee any order of elements. Theoretically it may happen that your file_list will look like `[('spam003', '.txt'), ('spam001', '.txt'), ('spam005', '.txt')]`. I recommend sorting it using built-in `sorted` function (returns a new list) or `list.sort` method (sorts in place, modify existing list). Luckily they are smart enough to deal with your list of tuples just out of the box :)

Reply:
fixed